scripts/scil_compute_metrics_stats_in_ROI.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Compute the statistics (mean, stddev) of scalar maps, which can represent
+diffusion metrics, in a ROI.
+
+The mask can either be a binary mask, or a weighting mask. If a weighting mask
+should either contain floats between 0 and 1, or should be normalized with
+--normalize_weights.
+
+IMPORTANT: if the mask contains weights (and not 0 and 1 exclusively), the
+standard deviation will also be weighted.
+"""
+
+import argparse
+import nibabel as nib
+import numpy as np
+import os
+import json
+
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_json_args,
+                             assert_inputs_exist)
+from scilpy.utils.filenames import split_name_with_nii
+from scilpy.utils.metrics_tools import get_roi_metrics_stats
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__,
+                                formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_mask',
+                   help='Mask volume file name, formatted in any nibabel ' +
+                        'supported format.\nCan be a binary mask or a ' +
+                        'weighted mask.')
+
+    g = p.add_mutually_exclusive_group(required=True)
+    g.add_argument('--metrics_dir',
+                   help='Metrics files directory. name of the directory ' +
+                        'containing the metrics files.')
+    g.add_argument('--metrics', dest='metrics_file_list', nargs='+',
+                   help='Metrics nifti file name. list of the names of the ' +
+                        'metrics file, in nifti format.')
+
+    p.add_argument('--bin', action='store_true',
+                   help='If set, will consider every value of the mask ' +
+                        'higher than 0 to be part of the mask, and set to 1 ' +
+                        '(equivalent weighting for every voxel).')
+
+    p.add_argument('--normalize_weights', action='store_true',
+                   help='If set, the weights will be normalized to the [0,1] '
+                        'range.')
+
+    add_overwrite_arg(p)
+    add_json_args(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, args.in_mask, [os.listdir(args.metrics_dir),
+                                               args.metrics_file_list])
+
+    # Load mask and validate content depending on flags
+    mask_img = nib.load(args.in_mask)
+
+    if not issubclass(mask_img.get_data_dtype().type, np.floating) and \
+            not args.normalize_weights:
+        parser.error('The mask file must contain floating point numbers.')
+
+    # Can be a weighted image
+    mask_data = mask_img.get_fdata(dtype=np.float32)
+
+    if args.normalize_weights:
+        mask_data /= np.sum(mask_data)
```

Comment:
Hum. So send to [0,1], shouldn't you divide by the max, not the sum? Am I mixed up?

Reply:
You're right it should not be sum (legacy code :weary: ). But `/ max` or `val - min / (max-min)` ? @mdesco

Reply:
I believe this is ok but would like to the guru to validate @jchoude.

Normalization, here, should not be taken in the usual way. The weighted mask is a track-count or tdi map normally. And the goal is to weigh the average FA by local TDI, i.e. FA weighs more where there are more tracks... So, / max or val - min / (max-min) is not what we want. I think it should be left as is...

Reply:
Here is an example:
```
import numpy as np
value = np.random.rand(10) # Imagine this is the FA (between 0 and 1)
weight = np.ones((10,)) # Imagine this is a streamline count (equal density everywhere)

# Our two possibilities
weight_1 = weight / np.max(weight)
weight_2 = weight / np.sum(weight)

np.average(value) # 0.431
np.average(value*weight_1)  # 0.431
np.average(value*weight_2) # 0.0431
```

Not sure the version #2 with the sum make sense, the more voxel you have the smaller your results get in relation to your original data while option #1 keep the same scaling.
(I personally use the max() version in scil_compute_connectivity())
```
if weighted:
    density = density / np.max(density)
```

@Arnaud, I thought about it and I think you should not do the -min() part because having a weight between 100 and 1000 would end up putting a 0 weight on the lowest part when in fact you really want to give it 10%. Also, simply dividing by the max() is negative value friendly !

Reply:
I agree the min-max normalisation is not really what we are looking for here but @mdesco dividing by sum does not seem to make sense to me. I would go for /max.

Waiting for @jchoude validation

Reply:
> You're right it should not be sum (legacy code 😩 ). But `/ max` or `val - min / (max-min)` ? @mdesco

If you decide to keep your change, careful, you wrote max instead of min `mask_data = mask_data - np.max(mask_data) / (np.max(mask_data) - np.min(max_data)) ` and also you are missing a parenthesis `(val - min) / (max-min)`

Reply:
Ok so. 

The reasoning of the sum is exactly what @mdesco said. Suppose you have a ROI and want to weigh each voxel differently, but still have a total sum of weights that go to 1. Then, you would need to divide by the sum, not by the max, to ensure that the sum of weights is one.

Now. I rechecked the old PRs, and dug down into my memory. I really remember that, at the time, there was a reason to do this, specifically. I can't seem to find it now, and I don't want to BS anything.

However: @frheault your example doesn't fit with the current code. When we call `np.average`, we specifically use the `weights` argument, which you don't. If I reuse your example with the correct syntax:

```
import numpy as np
value = np.random.rand(10) # Imagine this is the FA (between 0 and 1)
weight = np.ones((10,)) # Imagine this is a streamline count (equal density everywhere)

# Our two possibilities
weight_1 = weight / np.max(weight)
weight_2 = weight / np.sum(weight)

np.average(value) # 0.503
np.average(value, weights=weight_1)  # 0.503
np.average(value, weights=weight_2)  # 0.503
```

Notice the use of `weights` instead of multiplying directly. In your example, you were then missing a division by the sum of the weights.

If you compare the scaled ratios (`weight_1` and `weight_2`), you see that the weights are not the same, but their scaling to each other is the same (eg: `weight_1[0] / weight_1[1]` == `weight_2[0] / weight_2[1]`). This means that using either `sum` or `max` will yield the same (or almost the same, up to something like `10^-8`) end result.

So. It was not a bug before. If people really absolutely prefer using `max`, I'm ok with that for the sake of not having to redo all this discussion in a year or two. However, we shouldn't use min-max normalization, since it zeroes out the minimal values of the array. This is not a large error source, but not preferable.

Reply:
So for consistency, I will use that function in my scil_compute_connectivity.py (once merged)
Thanks for the explanation/correction!

Reply:
:+1: and it will be modified to use `max` I guess?

Reply:
I put max and add a link to this discussion into the code as comment. My guess, in a year or two we will ask our self the same question.